mongodbatlas/resource_mongodbatlas_search_index.go

```diff
@@ -231,11 +226,22 @@ func resourceMongoDBAtlasSearchIndexUpdate(ctx context.Context, d *schema.Resour
 		searchIndex.Analyzers = analyzers
 	}
 
+	if d.HasChange("mappings_dynamic") {
+		dynamic := d.Get("mappings_dynamic").(bool)
+		if searchIndex.Mappings == nil {
```

Comment:
for the edge case when the user tries to change from vectorSearch to search index